numpy/lib/npyio.py

```diff
@@ -86,10 +86,12 @@ def __dir__(self):
         return object.__getattribute__(self, '_obj').keys()
 
 
-def zipfile_factory(*args, **kwargs):
+def zipfile_factory(file, mode='r', *args, **kwargs):
```

Comment:
I'm not sure how in this case it would be a problem. Could you be more specific?

Before, it was just passing `*args` to `zipfile.Zipfile()`; the first two args to  `zipfile.Zipfile()` are necessarily `file` and `mode`, and AFAICT always have been. I'm not sure how this could be a backwards-compatibility problem.

Or are you thinking this might be a *future* compatibility problem?